pkg/ytconfig/generator.go

```diff
@@ -670,10 +677,19 @@ func (g *Generator) getHTTPProxyConfigImpl(spec *ytv1.HTTPProxiesSpec) (HTTPProx
 	g.fillCommonService(&c.CommonServer, &spec.InstanceSpec)
 	g.fillBusServer(&c.CommonServer, spec.NativeTransport)
 
-	if g.ytsaurus.Spec.OauthService != nil {
-		c.Auth.OauthService = ptr.To(getOauthService(*g.ytsaurus.Spec.OauthService))
-		c.Auth.OauthCookieAuthenticator = &OauthCookieAuthenticator{}
-		c.Auth.OauthTokenAuthenticator = &OauthTokenAuthenticator{}
+	oauthService := g.ytsaurus.Spec.OauthService
+	if oauthService != nil {
+		c.Auth.OauthService = ptr.To(getOauthService(*oauthService))
+		var createUserIfNotExist *bool
+		if oauthService.DisableUserCreation != nil {
+			createUserIfNotExist = ptr.To(!*oauthService.DisableUserCreation)
+		}
+		c.Auth.OauthCookieAuthenticator = &OauthCookieAuthenticator{
+			CreateUserIfNotExists: createUserIfNotExist,
+		}
+		c.Auth.OauthTokenAuthenticator = &OauthTokenAuthenticator{
+			CreateUserIfNotExists: createUserIfNotExist,
+		}
```

Comment:
Am I right that if `oauthService.DisableUserCreation == nil`, then `createUserIfNotExist` will stay `nil`, and `c.Auth.OauthCookieAuthenticator.CreateUserIfNotExists` will stay nil, and in resulting spec `create_user_if_not_exists` will not appear

Reply:
yes, that's the idea, on nil we just proxying default server behavior without making assumptions for default or forcing something.